controller/planner_backlog_blackbox_bench_test.go

```diff
@@ -0,0 +1,144 @@
+package controller_test
+
+import (
+	"testing"
+
+	"golang.org/x/net/context"
+
+	"github.com/almighty/almighty-core/account"
+	"github.com/almighty/almighty-core/app/test"
+	"github.com/almighty/almighty-core/application"
+	. "github.com/almighty/almighty-core/controller"
+	"github.com/almighty/almighty-core/gormapplication"
+	"github.com/almighty/almighty-core/gormsupport/cleaner"
+	gormbench "github.com/almighty/almighty-core/gormtestsupport/benchmark"
+	"github.com/almighty/almighty-core/iteration"
+	"github.com/almighty/almighty-core/log"
+	"github.com/almighty/almighty-core/migration"
+	"github.com/almighty/almighty-core/resource"
+	"github.com/almighty/almighty-core/space"
+	testsupport "github.com/almighty/almighty-core/test"
+	"github.com/almighty/almighty-core/workitem"
+
+	"github.com/goadesign/goa"
+	uuid "github.com/satori/go.uuid"
+)
+
+type BenchPlannerBacklogBlackboxREST struct {
+	gormbench.DBBenchSuite
+	clean        func()
+	testIdentity account.Identity
+	ctx          context.Context
+	svc          *goa.Service
+	ctrl         *PlannerBacklogController
+}
+
+var testBench *testing.T
+
+func TestRunPlannerBacklogBlackboxBenchREST(t *testing.T) {
+	testBench = t
+}
+
+func BenchmarkRunPlannerBacklogBlackboxREST(b *testing.B) {
+	resource.Require(b, resource.Database)
+	testsupport.Run(b, &BenchPlannerBacklogBlackboxREST{DBBenchSuite: gormbench.NewDBBenchSuite("../config.yaml")})
+}
+
+// SetupSuite overrides the DBTestSuite's function but calls it before doing anything else
+// The SetupSuite method will run before the tests in the suite are run.
+// It sets up a database connection for all the tests in this suite without polluting global space.
+func (rest *BenchPlannerBacklogBlackboxREST) SetupSuite() {
+	rest.DBBenchSuite.SetupSuite()
+	rest.ctx = migration.NewMigrationContext(context.Background())
+	rest.DBBenchSuite.PopulateDBBenchSuite(rest.ctx)
+	rest.svc = goa.New("PlannerBacklog-Service")
+	rest.ctrl = NewPlannerBacklogController(rest.svc, gormapplication.NewGormDB(rest.DB), rest.Configuration)
+}
+
+func (rest *BenchPlannerBacklogBlackboxREST) SetupBenchmark() {
+	rest.clean = cleaner.DeleteCreatedEntities(rest.DB)
+	// create a test identity
+	testIdentity, err := testsupport.CreateTestIdentity(rest.DB, "TestPlannerBacklogBlackboxREST user", "test provider")
+	if err != nil {
+		rest.B().Fail()
```

Comment:
@kwk i ignore the checks cause this benchmark aims to measure the performance. We rely on the unit and integration tests to validate the functionality. That is why i simply check if there are errors and then i fail the benchmark